src/tasks/MotionForceTask.cpp

```diff
@@ -164,6 +164,11 @@ void MotionForceTask::initialSetup() {
 		_current_position, _current_orientation, getLoopTimestep());
 	enableInternalOtgAccelerationLimited(0.3, 1.0, M_PI / 3, M_PI);
 
+	// singularity handling
+	setSingularityBounds(2e-2, 2e-3);  // default for panda 
+	MatrixXd J_posture = getConstRobotModel()->linkDependency(_link_name);
+	_singularity_handler = std::make_shared<SingularityHandler>(getConstRobotModel(), J_posture);
```

Comment:
Not too important for now, but maybe use a unique_ptr instead of a shared_ptr here